pkg/workflows/sdk/runtime.go

```diff
@@ -30,11 +30,13 @@ type Runtime interface {
 }
 
 type FetchRequest struct {
-	URL       string            `json:"url"`                 // URL to query, only http and https protocols are supported.
-	Method    string            `json:"method,omitempty"`    // HTTP verb, defaults to GET.
-	Headers   map[string]string `json:"headers,omitempty"`   // HTTP headers, defaults to empty.
-	Body      []byte            `json:"body,omitempty"`      // HTTP request body
-	TimeoutMs uint32            `json:"timeoutMs,omitempty"` // Timeout in milliseconds
+	URL         string            `json:"url"`                   // URL to query, only http and https protocols are supported.
+	Method      string            `json:"method,omitempty"`      // HTTP verb, defaults to GET.
+	Headers     map[string]string `json:"headers,omitempty"`     // HTTP headers, defaults to empty.
+	Body        []byte            `json:"body,omitempty"`        // HTTP request body
+	TimeoutMs   uint32            `json:"timeoutMs,omitempty"`   // Timeout in milliseconds
+	WithRetries *bool             `json:"withRetries,omitempty"` // Set to true to enable retry logic for requests
```

Comment:
Could this be `retries uint` where the default is 0 which represents do not retry?

Reply:
I agree that there's some duplication here, I think we can get away with just having MaxRetries below rather than both WithRetries and MaxRetries

Also as Justin points out, we could use 0 rather than a nullable type